internal/v2/s2av2.go

```diff
@@ -0,0 +1,72 @@
+package s2av2
+
+import (
+	"errors"
+	"context"
+
+	"google.golang.org/grpc/credentials"
+	"github.com/google/s2a-go/internal/v2/tlsconfigstore"
+	"github.com/google/s2a-go/internal/handshaker"
+)
+
+const (
+	s2aSecurityProtocol = "s2av2"
+)
+
+type s2av2TransportCreds struct {
+	info     *credentials.ProtocolInfo
+	config   *tls.Config
+	isClient bool
+}
+
+func NewClientCreds() (*credentials.TransportCredentials) {
+	return &s2av2TransportCreds{
+		info: &credentials.ProtocolInfo{
+			SecurityProtocol: s2aSecurityProtocol,
+		},
+		config: tlsconfigstore.GetTlsConfigurationForClient(),
+		isClient: true,
+	}
+}
+
+func NewServerCreds() *credentials.TransportCredentials {
+	return &s2av2TransportCreds{
+		info: &credentials.ProtocolInfo{
+			SecurityProtocol: s2aSecurityProtocol,
+		},
+		config: tlsconfigstore.GetTlsConfigurationForServer(),
+		isClient: false,
+	}
+}
+
+func (c *s2av2TransportCreds) ClientHandShake(ctx context.Context, serverAuthority string, rawConn net.Conn) (_ net.Conn, _ credentials.AuthInfo, err error) {
```

Comment:
The function name needs to exactly match the function name in the interface, so in this case it should be `ClientHandshake`.